plugins/Flanger/QuadratureLfo.h

```diff
@@ -30,51 +30,66 @@
 class QuadratureLfo
 {
 public:
-	QuadratureLfo( int sampleRate );
+	QuadratureLfo( int sampleRate )
+	{
+		setSampleRate(sampleRate);
+	}
 	~QuadratureLfo()
 	{
 	}
 
 	inline void setFrequency( double frequency )
 	{
-		if( frequency < 0 || frequency > ( m_samplerate / 2.0 ) || frequency == m_frequency )
+		if( frequency < 0 || frequency > m_srOverTwo || frequency == m_frequency )
 		{
 			return;
 		}
 		m_frequency = frequency;
 		m_increment = m_frequency * m_twoPiOverSr;
 
-		if( m_phase >= F_2PI )
+		if( m_phase >= D_2PI )
 		{
-			m_phase -= F_2PI;
+			m_phase -= D_2PI;
 		}
 	}
 
 
-
-
 	inline void restart()
 	{
 		m_phase = 0;
 	}
 
 
-
-
 	inline void setSampleRate ( int samplerate )
 	{
 		m_samplerate = samplerate;
 		m_twoPiOverSr = F_2PI / samplerate;
+		m_srOverTwo = samplerate / 2.0;
 		m_increment = m_frequency * m_twoPiOverSr;
 	}
 
-	void tick( float *s, float *c );
+
+	inline void setOffset( double offsetVal )
+	{
+		m_offset = offsetVal;
+	}
+
+
+	void tick( float *s, float *c )
+	{
+		*s = sinf( m_phase );
+		*c = sinf( m_phase + m_offset );
+		m_phase += m_increment;
+
+	}
 
 private:
 	double m_frequency;
 	double m_phase;
 	double m_increment;
 	double m_twoPiOverSr;
+	double m_srOverTwo;
```

Comment:
Do we really need this variable?

Reply:
You're right, my brain saw that and saw am unnecessarily small performance enhancement but it failed to realize that it's only when the setFrequency function is called, which probably won't be called every loop anyway... I think I did it because m_twoPiOverSr appears to exist for the same reason.  I'll revert it.
(And while I'm at it, I'll make it so setFrequency isn't called every loop in the Phaser, I didn't catch that...)